src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -56,32 +56,32 @@ protected Future<Void> processInventoryCreationEvent(ResourceEvent resourceEvent
     var permanentLocationId = holdingObject.getString(PERMANENT_LOCATION_ID.getValue());
     var tenantIdFromEvent = resourceEvent.getTenant();
     return dbClient.getPgClient()
-      .withTrans(conn -> {
+      .withTrans(conn -> createDBClient(tenantIdFromEvent).getPgClient().withTrans(connAudit -> {
```

Comment:
if it 2 different sql connection - the same transaction is not possible that can use 2 connections.

Reply:
it seams need some refactoring to save audit logs separately in scope of another connection without transaction.
If it is complex lets merge all other optimization at least to pass karate tests, and refactoring in scope of another PR

Reply:
> it seams need some refactoring to save audit logs separately in scope of another connection without transaction. If it is complex lets merge all other optimization at least to pass karate tests, and refactoring in scope of another PR

Will try refactoring now